Assets/Scripts/Bomb.cs

```diff
@@ -44,6 +49,9 @@ public class Bomb: MonoBehaviour
 
         explodeBombEvent.Raise();
 
+        // this call must happen after `DestroyTerrain()` 
+        detonate?.Invoke();
+
```

Comment:
We should definitely just use one event system, :P. I'll take a look at using this in the future.

Reply:
Yea, or I just need help understanding how I can hook into yours.  Let's maybe pair up this weekend?